server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/TransportListTasksAction.java

```diff
@@ -66,19 +84,68 @@ protected void taskOperation(Task actionTask, ListTasksRequest request, Task tas
     }
 
     @Override
-    protected void processTasks(ListTasksRequest request, Consumer<Task> operation) {
+    protected void processTasks(ListTasksRequest request, Consumer<Task> operation, Runnable nodeOperation, Consumer<Exception> onFailure) {
```

Comment:
We only ever run one of `nodeOperation` or `onFailure` so IMO this makes more sense as an `ActionListener<Void>`.